### DISCUSSION TITLE
Last Feed Time vs Last Time First Fed

### DISCUSSION THREAD

**Discussion by sstratoti** [babybuddy/babybuddy#306] [Ideas]
Our lactation consultant and a few videos we’ve watched have told us that for a new born we shouldn’t be going between 2-3 hours between feedings based on the last time we first started feeding. 

Currently we’re doing 15 left breast, 15 right breet then 1oz formula until my wife’s milk comes in. But we’ve been incorrectly been going off of the “last feeding time” or “fed x hours ago” which I believe is based off of the end time, not the start time. Moreover, since our feedings result in 3 logs, this means that it’s going off of the last time our baby was successfully fed, which could take some time to latch between breasts. 

Perhaps you could do a query to find the last 10 entries, and determine if a cluster exists? (Within 30 mins of each other or something) then have this dashboard item reflect the start time of that feeding? Or make it configurable to display one or the other? We’ve been accidentally spacing out our baby’s feeding times as a result. 

Just a suggestion!

**Comment by cdubz** [comment]
The Last Feeding card is based off the _start_ time of the last feeding. It’s tough to use well early on but our approach has generally been to record the entire duration of the feeding. I.e. we start a timer when the feeding starts, switch breasts, burp, warm more milk if bottle, etc. and then only stop it when all of those activities are done. And we record the _last_ breast side used.

The important data points, as we understand it, are the start time and last breast side because that tells us the time between feedings and which breast to start on next. That is, it’s not necessarily important to record each breast separately (or at least I’ve never heard of a use case for that). Do y’all have a reason to record them separately that I’m not thinking about?

**Comment by sstratoti** [comment]
My wife had a breast reduction surgery and they tried to keep her nipples attached. It’s helpful to track what side she is latching on to for longer for the lactation consultant to review and maybe figure out what we’re doing wrong. Our baby tends to favor her left side for some reason. It could be the hold or the angle, not sure. The doctor also asked us to try to do 15 mins on each side, then formula. So it’s helpful to time things as well. 

So for the feeding you track the entire duration? Even if the baby cries / needs soothing, or doesn’t latch right away?

**Comment by cdubz** [comment]
Yeah, but we have never had a real need to be super accurate about the timing — our key day points have always been last breast and duration from start. It sounds like you do want timing. For keeping track of the 15 mins. you could theoretically start one timer for the full feeding and a second and third (that you don’t record) for each breast. But that wouldn’t help you with evaluating the timing for each breast.

Building in “cluster detection” seems technically straightforward but I would worry about the UX side of it. When and how do you explain to the user how it works? And what if it is undesired behavior for other use cases (not that I could think of any...)?

**Comment by sstratoti** [comment]
Cluster feedings are the worrrrrst. 
I guess when I saw “type” and “method”, I was thinking each was its own log. Especially if you want to track that you fed formula to “top them off”. I agree, the UI could be tricky. But instead of having to add multiple timers to a single feed log, couldn’t you group them by proximity in the baby timeline? Or in the dashboard for “last feeding” group them by proximity to other entries and give the earliest time?

Right now we’re just setting a personal timer on our phones for 3 hours from the last feeding start, in addition to starting a new timer for each feed action. We think that’s working well. Just wanted to spitball some ideas while I’m lucid. :)
